### PR TITLE
Convert Filebeat haproxy.log to ECS even more!

### PR DESCRIPTION
**Pull request by webmat**
This module was migrated prior to ECS 1.0.0-beta2, and therefore has a few things left to migrate.

**Caveats**

- `haproxy.bytes_read`: Both the TCP logs and the HTTP logs use the field. So it's copied to `http.response.bytes` when parsing HTTP logs, but not when parsing TCP logs.

**Renames**

- haproxy.http.response.status_code => http.response.status_code

Different scale, original fields still present:

- haproxy.http.request.time_active_ms => event.duration
- haproxy.tcp.processing_time_ms => event.duration

Note that `haproxy.client.ip` is migrated in #10141

**TODO**

- [x] Alias renamed fields to their ECS counterpart, not forgetting `migration: true`
- [x] Document field migrations in ecs-migration.yml
- [x] Changelog

### CODE REVIEW

File: filebeat/module/haproxy/log/test/httplog-no-headers.log-expected.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     {
         "ecs.version": "1.0.0-beta2",
         "event.dataset": "haproxy.log",
+        "event.duration": 0,
```
Comment (ruflin) [inline filebeat/module/haproxy/log/test/httplog-no-headers.log-expected.json]: Is this 0 if it does not exist or is it actually 0 in the log?

File: filebeat/module/haproxy/log/test/tcplog.log [hunk 2]
```diff
@@ -1 +1 @@
-Sep 20 15:44:23 127.0.0.1 haproxy[25457]: 127.0.0.1:40962 [20/Sep/2018:15:44:23.285] main app/<NOSRV> -1/-1/0 212 SC 1/1/0/0/0 0/0
+Sep 20 15:44:23 127.0.0.1 haproxy[25457]: 127.0.0.1:40962 [20/Sep/2018:15:44:23.285] main app/<NOSRV> -1/-1/1 212 SC 1/1/0/0/0 0/0
```
Comment (ruflin) [inline filebeat/module/haproxy/log/test/tcplog.log]: We should always use at least 2 log lines to make sure things work as expected. Not a big issue here as we don't have multiline.

File: filebeat/module/haproxy/log/test/httplog-no-headers.log-expected.json (hunk 1 above)
Comment (webmat) [reply filebeat/module/haproxy/log/test/httplog-no-headers.log-expected.json]: It's always present in this log format. It's the last 0 of `0/-1/-1/-1/0 ` in the first line, just before the 503 status code.

File: filebeat/module/haproxy/log/test/tcplog.log (hunk 2 above)
Comment (webmat) [reply filebeat/module/haproxy/log/test/tcplog.log]: Agreed

**Comment by webmat** [comment]
Spotted one detail prior to merging. Will merge when CI is green.

**Comment by webmat** [comment]
CI test failures unrelated

**Pull request merged**